### PR TITLE
problem 2 recursive solution

### PR DESCRIPTION
**Pull request by LastPossum**
Here is recursive solution for problem2 with tests

### CODE REVIEW

File: others/problem-2/solution.go [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+	//knapSack is full or there is no new items
+	if w <= 0 || len(wt) == 0 || len(val) == 0 {
+		return 0
+	}
+	//if item is to heavy - pass
+	if wt[0] > w {
+		return knapSack(w, wt[1:], val[1:])
+	}
+
+	// return the maximum of two cases:
```
Comment (princerachit) [inline others/problem-2/solution.go:16]: good job but can you make variable name more intuitive?

Comment (princerachit) [reply others/problem-2/solution.go:16]: var1 and var2 is not the best naming convention

**Review by princerachit — CHANGES_REQUESTED** [review]

File: others/problem-2/solution.go (hunk 1 above)
Comment (LastPossum) [reply others/problem-2/solution.go:16]: There are 2 hard problems in computer science: cache invalidation, naming things

I'll change these names. But what is the best choice? I suggest addFirstItem and dropFirstItem for var1 and var2 respectively, what do you think?

Comment (princerachit) [reply others/problem-2/solution.go:16]: True.

**Review by princerachit — APPROVED** [review]
Thanks for the amazong pr. You can create issues to add more problems and the wrote solutions for it. Just let me know so i can add tags to them.

**Comment by princerachit** [comment]
Fixes https://github.com/princerachit/cracking-the-coding-interview-4th-edition/issues/8

**Pull request merged**